packages/graphql-typescript-definitions/src/print/schema/index.ts

```diff
@@ -35,13 +39,34 @@ export function printSchema(schema: GraphQLSchema, options: Options = {}) {
       continue;
     }
 
-    fileBody.push(
-      t.exportNamedDeclaration(tsTypeForRootInputType(type, options), []),
-    );
+    if (isEnumType(type)) {
+      const enumType = tsEnumForType(type, options);
+      fileMap.set(
+        `${enumType.id.name}.ts`,
+        generate(
+          t.file(t.program([t.exportNamedDeclaration(enumType, [])]), [], []),
+        ).code,
+      );
+      fileBody.unshift(
+        t.importDeclaration(
+          [t.importSpecifier(enumType.id, enumType.id)],
+          t.stringLiteral(`./${enumType.id.name}`),
+        ),
+        t.exportNamedDeclaration(null, [
```

Comment:
I think we can probably `export {default as NAME} from './NAME';` here?

Reply:
I don't think you can `export default` for an enum in a single statement. It's not to say we couldn't create two statements in each enum module, but it seemed unnecessary to add the complexity.

Reply:
Yeah there should just be one line for the export (whatever that needs to be, default export makes most sense but doesn't matter much), and one for the combined import/ export in the index (more what I was getting at with my comment)

Reply:
ah for the single export in the index module. The issue there is that some types may need to reference some of the enum types, so we need to import them first.

Reply:
Ah ok, makes sense